plugins/ssh-plugin/src/ssh-plugin-backend.ts

```diff
@@ -45,76 +54,93 @@ export async function start() {
+    const action = await theia.window.showInformationMessage('Key pair successfully generated, do you want to view the public key?', viewAction);
     if (action === viewAction && key.privateKey) {
-        theia.workspace.openTextDocument({ content: key.privateKey });
+        const document = await theia.workspace.openTextDocument({ content: key.publicKey });
+        await theia.window.showTextDocument(document);
+    }
+};
+
+const generateKeyPairForHost = async function (sshkeyManager: SshKeyManager): Promise<void> {
+    const hostName = await getHostName();
+    const key = await sshkeyManager.generate('vcs', hostName);
+    updateConfig(hostName);
+    writeKey(hostName, key.privateKey);
+    const viewAction = 'View';
+    const action = await theia.window.showInformationMessage(`Key pair for ${hostName} successfully generated, do you want to view the public key?`, viewAction);
+    if (action === viewAction && key.privateKey) {
+        const document = await theia.workspace.openTextDocument({ content: key.publicKey });
+        await theia.window.showTextDocument(document);
     }
 };
 
 const createKeyPair = async function (sshkeyManager: SshKeyManager): Promise<void> {
-    const keyName = await theia.window.showInputBox({ placeHolder: 'Please provide a key pair name' });
+    const hostName = await getHostName();
     const publicKey = await theia.window.showInputBox({ placeHolder: 'Enter public key' });
+    const privateKey = await theia.window.showInputBox({ placeHolder: 'Enter private key' });
 
     await sshkeyManager
-        .create({ name: keyName ? keyName : '', service: await getSshServiceName(), publicKey: publicKey })
-        .then(() => {
-            theia.window.showInformationMessage('Key "' + `${keyName}` + '" successfully created');
+        .create({ name: hostName, service: 'vcs', publicKey: publicKey, privateKey })
+        .then(async () => {
+            theia.window.showInformationMessage('Key "' + `${hostName}` + '" successfully created');
+            updateConfig(hostName);
+            writeKey(hostName, privateKey);
         })
         .catch(error => {
             theia.window.showErrorMessage(error);
         });
 };
 
 const deleteKeyPair = async function (sshkeyManager: SshKeyManager): Promise<void> {
-    const sshServiceName = await getSshServiceName();
-    const keys: cheApi.ssh.SshPair[] = await sshkeyManager.getAll(sshServiceName);
+    const keys: cheApi.ssh.SshPair[] = await sshkeyManager.getAll('vcs');
     const keyResp = await theia.window.showQuickPick<theia.QuickPickItem>(keys.map(key =>
         ({ label: key.name ? key.name : '' })), {});
     const keyName = keyResp ? keyResp.label : '';
     await sshkeyManager
-        .delete(sshServiceName, keyName)
+        .delete('vcs', keyName)
         .then(() => {
+            const keyFile = os.homedir() + '/.ssh/' + keyName.replace('.', '_');
+            if (fs.existsSync(keyFile)) {
```

Comment:
avoid any sync, we're inside async functions

Reply:
done